src/librc/librc.c

```diff
@@ -587,6 +589,9 @@ free_rc_dirs(void)
 	free(rc_dirs.runleveldir);
 	rc_dirs.runleveldir = NULL;
 	free(rc_dirs.svcdir);
+	for (size_t i = 0; i < ARRAY_SIZE(scriptdirs); i++)
+		if (rc_dirs.scriptdirs_data[i])
+			free(rc_dirs.scriptdirs_data[i]);
```

Comment:
Check is pretty useless since `free(NULL)` is no-op. I'd also set it back to NULL after `free()`, that's what the rest is doing.